src/main/java/uk/gov/hmcts/darts/event/service/impl/CleanupCurrentFlagEventProcessorImpl.java

```diff
@@ -19,36 +19,24 @@
 @Getter
 @RequiredArgsConstructor
 @Slf4j
+@Component
 public class CleanupCurrentFlagEventProcessorImpl implements CleanupCurrentFlagEventProcessor {
-    private final Integer batchSize;
     private final EventRepository eventRepository;
     private final HearingRepository hearingRepository;
-    private final UserIdentity userIdentity;
 
     @Override
-    public List<Integer> processCurrentEvent() {
-        List<Integer> processedEventIdLst = new ArrayList<>();
-        log.debug("Batch size to process event ids for {}", batchSize);
-        List<Integer> eventEntityReturned = eventRepository.findCurrentEventIdsWithDuplicates(batchSize);
-        if (log.isDebugEnabled()) {
-            log.debug("Event ids being processed {}", eventEntityReturned
-                .stream()
-                .map(Object::toString)
-                .collect(Collectors.joining(",")));
+    @Async("eventTaskExecutor")
+    public void processEvent(Integer eventId) {
+        if (eventId == null || eventId == 0) {
+            return;
         }
+        log.info("Cleaning up event id {}", eventId);
```

Comment:
Could we improve the log message to say what we are doing? "Cleaning up" could be interpreted as removing the PII which is part of a different job.

Reply:
Updated